src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -856,12 +856,29 @@ private void GeneratePropertySetter<T>(QualifiedType returnType, T decl, Class @
                 WriteStartBraceIndent();
 
                 var marshal = new CSharpMarshalManagedToNativePrinter(ctx);
-                ctx.ReturnVarName = string.Format("{0}{1}{2}",
+
+                var arrayType = field.Type as ArrayType;
+                
+                if(arrayType!=null && @class.IsValueType==true)
```

Comment:
Please put spaces around '!=' and do not use ==/!= for booleans: just use if (var)/if (!var).